src/js/ttsHookup.js

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2015 OCAD University
+
+Licensed under the Educational Community License (ECL), Version 2.0 or the New
+BSD license. You may not use this file except in compliance with one these
+Licenses.
+
+You may obtain a copy of the ECL 2.0 License and BSD License at
+https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
+*/
+
+(function ($, fluid) {
+
+    "use strict";
+
+    fluid.registerNamespace("gpii.firstDiscovery");
+
+    fluid.defaults("gpii.firstDiscovery.tts.hookup", {
+        invokers: {
+            speak: {
+                func: "{fluid.textToSpeech}.queueSpeech"
+            }
+        }
+    });
+
+    fluid.registerNamespace("gpii.firstDiscovery.tts.tooltipHookup");
+
+    fluid.defaults("gpii.firstDiscovery.tts.tooltipHookup", {
+        gradeNames: ["gpii.firstDiscovery.tts.hookup"],
+        listeners: {
+            "afterOpen.tts": {
+                listener: "gpii.firstDiscovery.tts.tooltipHookup.speakTooltip",
+                args: ["{that}", "{arguments}.2"]
+            }
+        }
+    });
+
+    gpii.firstDiscovery.tts.tooltipHookup.speakTooltip = function (that, tooltip) {
+        that.speak(tooltip.text());
+    };
+
+    fluid.registerNamespace("gpii.firstDiscovery.tts.fdHookup");
+
+    // TODO: Currently this hookup is intended to be added to gpii.firstDiscovery.firstDiscoveryEditor
+    // directly. However, it should be reconfigured like the tooltip hookup and placed at the panel level.
+    // The issue at the moment is that a given panel doesn't know when this it is visible. All this information
+    // is contained at the editor level, which also doesn't really know which panel component is shown.
+    fluid.defaults("gpii.firstDiscovery.tts.fdHookup", {
+        components: {
+            selfVoicing: {
+                options: {
+                    gradeNames: ["fluid.prefs.msgLookup"],
+                    members: {
+                        messageResolver: "{firstDiscoveryEditor}.msgResolver"
+                    },
```

Comment:
Line 52 - 55 are not needed since they've defined in `gpii.firstDiscovery.firstDiscoveryEditor` for `selfVoicing`.